### DISCUSSION TITLE
Change partition name using .overlay

### DISCUSSION THREAD

**Discussion by joaodullius** [zephyrproject-rtos/zephyr#37545] [Q&A]
I'm trying to add mcuboot support to arduino_nano_33_ble using the following arduino_nano_33_ble.overlay file in the boards directory in my project:

```
 / {
	  chosen {
		  zephyr,code-partition = &slot0_partition;
	  };
  };

&flash0 {

	partitions {
		compatible = "fixed-partitions";
		#address-cells = <1>;
		#size-cells = <1>;

		boot_partition: partition@10000 {
			label = "mcuboot";
			reg = <0x000010000 0x0000C000>;
		};
		slot0_partition: partition@1c000 {
			label = "image-0";
			reg = <0x0001C000 0x00067000>;
		};
		slot1_partition: partition@83000 {
			label = "image-1";
			reg = <0x00083000 0x00067000>;
		};
		scratch_partition: partition@ea000 {
			label = "image-scratch";
			reg = <0x000ea000 0x0001e000>;
		};

		/*
		 * The flash starting at 0x000f8000 and ending at
		 * 0x000fffff is reserved for use by the application.
		 */

		/*
		 * Storage partition will be used by FCB/LittleFS/NVS
		 * if enabled.
		 */
		storage_partition: partition@f8000 {
			label = "storage";
			reg = <0x000f8000 0x00008000>;
		};
	};
};
```

This is the partitions defined on the board dts:
```
&flash0 {

	partitions {
		compatible = "fixed-partitions";
		#address-cells = <1>;
		#size-cells = <1>;

		code_partition: partition@10000 {
			label = "code";
			reg = <0x10000 0xf0000>;
			read-only;
		};
	};
};
```

West is correctly applying the above overlay, but it is mixing with the original partition definition, as we can see on the generated build/zephyr/zephyr.dts :

```
flash0: flash@0 {

				compatible = "soc-nv-flash";
				label = "NRF_FLASH";
				erase-block-size = < 0x1000 >;
				write-block-size = < 0x4 >;
				reg = < 0x0 0x100000 >;
				partitions {
					compatible = "fixed-partitions";
					#address-cells = < 0x1 >;
					#size-cells = < 0x1 >;
					code_partition: boot_partition: partition@10000 {
						label = "mcuboot";
						reg = < 0x10000 0xc000 >;
						read-only;
					};
```

* notice the double named partition code_partition; coot_partition: partition@10000 above

What would be the correct way to remove the original partition name?

**Comment by Laczen** [comment]
@joaodullius, you should use delete-node to remove nodes from the original dts:
```
/delete-node/ &boot_partition;
...
```

**Reply by joaodullius** [reply]
Hi @Laczen,

As the partition I'm trying to delete is originaly named "code_partition" I tried the following:

'''
&flash0 {

	partitions {
		compatible = "fixed-partitions";
		#address-cells = <1>;
		#size-cells = <1>;

		/delete-node/ &code_partition;

		boot_partition: partition@10000 {
			label = "mcuboot";
			reg = <0x000010000 0x0000C000>;
		};
		...
'''
		
But I get the following error when adding this line:
-- Found devicetree overlay: C:/Work/nrf/v1.6.1/zephyr/samples/bluetooth/peripheral/boards/arduino_nano_33_ble.overlay
Error: arduino_nano_33_ble.dts.pre.tmp:484.17-32 syntax error
FATAL ERROR: Unable to parse input tree

Update:
I also tried with "/delete-node/ code_partition;" (no &). 
I get no errors but the partition is still double named on the zephyr.dts

**Reply by Laczen** [reply]
@joaodullius, put the delete-node line before &flash0, and keep the &.

**Reply by joaodullius** [reply]
@Laczen 
Thanks, I was able to remove the old "code-partition" by adding "/delete-node/ &code_partition;" before &flash0.

But now, I'm getting a different error:

-- Found BOARD.dts: C:/Work/nrf/v1.6.1/zephyr/boards/arm/arduino_nano_33_ble/arduino_nano_33_ble.dts
-- Found devicetree overlay: C:/Work/nrf/v1.6.1/bootloader/mcuboot/boot/zephyr/dts.overlay
<stdout>: ERROR (path_references): /chosen: Reference to non-existent node or label "boot_partition"

ERROR: Input tree has errors, aborting (use -f to force output)

It appears that mcuboot dts.overlay can´t locate the boot_partition.
I checked and it is present at zephyr.dts:
![image](https://user-images.githubusercontent.com/8560057/128738954-3b4239ab-d956-4401-ae64-18a1c803ca19.png)
`

**Reply by Laczen** [reply]
Sometimes it helps to do a pristine build, have you tried that? Esspecially when changing overlay files I always use pristine build.

**Reply by joaodullius** [reply]
Hi @Laczen ,

Yes, I tried with -p and also deleted the build directory, same error.

**Reply by Laczen** [reply]
Hi @joaodullius, forgotten the & in `zephyr, code-partition = &boot_partition;` ?
